### ISSUE TITLE
"bosh download manifest" returns file that is different from its manifest

### ISSUE THREAD

**Issue by liuweichu** [cloudfoundry/bosh#1530]
We are on BOSH director v260.

Our manifest has all releases version and stemcell version as integer. For example 

```
- name: errand
  cloud_properties:
    availability_zone: ...
    instance_type: ...
  network: ...
  stemcell:
    name: bosh-openstack-kvm-ubuntu-trusty-go_agent
    version: 3215
```

However what we get from `bosh download manifest`, it becomes

```
- name: errand
  ...
  stemcell:
    name: bosh-openstack-kvm-ubuntu-trusty-go_agent
    version: '3215'
```

The same conversion is also seen at release version.

Such implicit conversion is confusing, because output manifest is not the same as input anymore.

**Comment by liuweichu** [comment]
Even it's decided that BOSH doesn't accept integer version anymore (which I tried to search through issues but didn't find), it should be that giving warning at `bosh deployment xx.yml`, rather than converting it implicitly

**Comment by cppforlife** [comment]
@liuhewei will take a look. we typically just save entire contents of manifest as is and return that to you. not sure why it would behave differently in this case. do you know if this problem also affects new CLI (https://github.com/cloudfoundry/bosh-cli)?

**Comment by liuweichu** [comment]
@cppforlife Hi, I think this is a BOSH director side behavior.
When it parses the manifest, release/stemcell versions are implicitly converted to string version in BOSH, no matter what the input was.

new-cli is shows the same behavior.

```
# version is 9 and 10 instead of '9' and '10' in bosh.yml
$ bosh-gocli deploy bosh.yml
Using environment 'http://192.168.50.4:25555' as user 'admin'

Using deployment '...'

  releases:
  - name: ...
-   version: '9'
+   version: '10'
```

My `bosh-lite` director version is `1.3262.3.0`, so this has been around for a while. (We just noticed this because we bumped our director recently from v236 to v260)
If this is the expected behavior of BOSH, it's worthy to be mentioned in BOSH doc because input manifest≠output manifest is not quite intuitive

**Comment by mfine30** [comment]
It looks like this issue has since been resolved as of Director v266.2.0. If it continues to be a concern for you, please let us know.
